call-home/src/common/iterator.rs

```diff
@@ -0,0 +1,69 @@
+#[derive(Debug, Clone)]
+pub enum SortOrder {
+    NonDescending,
+    NonAscending,
+}
+
+#[derive(Debug, Clone)]
+pub struct SortedExactSizeIter<T> {
+    data: Vec<T>,
+    sort_order: SortOrder,
+}
+
+impl<T: Ord> SortedExactSizeIter<T> {
+    // Create a new SortedExactSizeIter from an iterator and a sort order.
+    pub fn from_iter<I>(iter: I, sort_order: SortOrder) -> Self
+    where
+        I: Iterator<Item = T>,
+    {
+        let mut data: Vec<T> = iter.collect();
+        match sort_order {
+            SortOrder::NonDescending => data.sort_unstable(),
```

Comment:
unstable?

Reply:
It's faster: https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable
Stable is not necessary for us. It would make sense to use stable sort, if the move operations were very costly. But they are constant time and cheap.https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable

Stable here talks about if it will reorder elements or not. Not an indicator of the stability of the code.